### ISSUE TITLE
LoadLibrary for Windows + Python 3.8 doesn't seem to search through PREFIX/Library/bin

### ISSUE THREAD

**Issue by hmaarrfk** [conda-forge/python-feedstock#307]
It seems that on Windows + Python 3.8 LoadLibrary misses to search through `Library/bin`

This could be due to 
> DLL dependencies for extension modules and DLLs loaded with ctypes on Windows are now resolved more securely. Only the system paths, the directory containing the DLL or PYD file, and directories added with add_dll_directory() are searched for load-time dependencies. Specifically, PATH and the current working directory are no longer used, and modifications to these will no longer have any effect on normal DLL resolution. If your application relies on these mechanisms, you should check for add_dll_directory() and if it exists, use it to add your DLLs directory while loading your library. Note that Windows 7 users will need to ensure that Windows Update KB2533623 has been installed (this is also verified by the installer). (Contributed by Steve Dower in bpo-36085.)

https://docs.python.org/3.8/whatsnew/3.8.html#changes-in-the-python-api

https://dev.azure.com/ramonaoptics/feedstock-builds/_build/results?buildId=2277&view=logs&j=e5cdccbf-4751-5a24-7406-185c9d30d021&t=d66ed8ca-07ff-5419-c5d1-5e21fa500bb4&l=843

Issue:

<br/>
Environment (<code>conda list</code>):
<details>

```
The following NEW packages will be INSTALLED:

    ca-certificates:         2019.11.28-hecc5488_0  conda-forge
    certifi:                 2019.11.28-py38_0      conda-forge
    libprotobuf:             3.10.1-h1a1b453_0      conda-forge
    m2w64-gcc-libgfortran:   5.3.0-6                defaults   
    m2w64-gcc-libs:          5.3.0-7                defaults   
    m2w64-gcc-libs-core:     5.3.0-7                defaults   
    m2w64-gmp:               6.1.0-2                defaults   
    m2w64-libwinpthread-git: 5.0.0.4634.697f757-2   defaults   
    msys2-conda-epoch:       20160418-1             defaults   
    openssl:                 1.1.1d-hfa6e2cd_0      conda-forge
    protobuf:                3.10.1-py38he025d50_0  conda-forge
    python:                  3.8.1-he1f5543_1       conda-forge
    rx:                      1.6.1-py_0             conda-forge
    setuptools:              45.0.0-py38_1          conda-forge
    six:                     1.14.0-py38_0          conda-forge
    sqlite:                  3.30.1-hfa6e2cd_0      conda-forge
    vc:                      14.1-h0510ff6_4        defaults   
    vs2015_runtime:          14.16.27012-hf0eaf9b_1 defaults   
    wincertstore:            0.2-py38_1003          conda-forge
    zaber-motion:            1.0.8-py38h6f9f611_2   local      
    zlib:                    1.2.11-h2fa13f4_1006   conda-forge

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

```
</details>

**Comment by mingwandroid** [comment]
Well we've undone this change with Steve Dower's blessing so something else must be up with this stuff.

I think the bug is likely in pywin32 to be honest with you. We fixed this on defaults recently, I think @msarahan made a PR to conda-forge for this?

**Comment by hmaarrfk** [comment]
Thanks for the reply.

Are you referring to https://github.com/conda-forge/pywin32-feedstock/pull/20

**Comment by mingwandroid** [comment]
Ah, no, I don't see the PR, I may have imagined it! Taking a quick look at resyncing now.

**Comment by mingwandroid** [comment]
https://github.com/conda-forge/pywin32-feedstock/pull/21

**Comment by mingwandroid** [comment]
Copying MS headers here. Asking Steve Dower if this is OK. If not, we'll need another approach. The most important part is https://github.com/conda-forge/pywin32-feedstock/blob/6d5340c2c00cc6a82394fee9fdb597f24766248c/recipe/0003-remove-PATH-pth-hack.patch and I'm happy to drop the exhange changes if it is a deal-breaker.

**Comment by hmaarrfk** [comment]
don't think i have pywin32 in my build.

My workaround, was to call `libname = ctypes.util.find_library(libname)`

```
+    lib_name = "zaber-motion-lib".format(ext)
+    # On windows, since Python 3.8 you need an extra call to find_library
+    # https://github.com/beetbox/pyacoustid/issues/54
+    if sys.platform == 'win32':
+        lib_name = ctypes.util.find_library(lib_name)
+    return cdll.LoadLibrary(lib_name)
+    return cdll.LoadLibrary(lib_name)
```

**Comment by jakirkham** [comment]
Yeah we had to do something similar with `llvmlite` as well ( https://github.com/conda-forge/llvmlite-feedstock/pull/44 ).

cc @mbargull

**Comment by hmaarrfk** [comment]
@jakirkham should you move your patch to this repository? It seems like the more correct way to do things.

**Comment by jakirkham** [comment]
It’s not really my patch. We have @mbargull to thank for it 🙂

No objection to it living here. Though maybe there’s a better way to solve it in `python`?

**Comment by isuruf** [comment]
Fixed in 3.9 and 3.10

**Comment by jakirkham** [comment]
That's great news! 😄

Should we keep this open though as the issue was about Python 3.8 or was it fixed there as well?
